woocommerce/woocommerce-android
# [REST API] Password screen for the Jetpack connection

Closes: #8396 

### Description
This PR adds the password screen for the WordPress.com authentication needed for the Jetpack connection.
For now, this supports only the basic flow of "email + password" login, so no support yet for:
1. 2FA
2. Magic link

### Testing instructions
1. Open the app, then sign in using a non-Jetpack site.
2. Click on the Jetpack benefits banner.
3. Click on "Log in to continue"
4. Enter a valid WordPress.com email, then click on "Install Jetpack"
5. Enter a wrong password then click on continue on "Install Jetpack"
6. Confirm an error is shown.
7. Enter the correct password.
8. Confirm the Jetpack installation/connection starts (the flow will fail after finishing the connection, this will be handled in future PRs).

Note: if your account has 2FA enabled, at step 8, a Toast will be shown instead.

### Images/gif
<img width="360" src="https://user-images.githubusercontent.com/1657201/221632991-e6981d8e-33bc-4dbc-9eb8-a609a246034d.png" />
